### PR TITLE
fixing the release

### PR DESCRIPTION
**Pull request by MelSumner**
in #1397 a user has reported that types are missing and they are getting an error. 

This PR updates the `.npmignore`  file and the package.json file to more correctly reflect what should be getting published re:types. 

This error seems to be related to the differences between `npm pack` and `yarn pack`.

Also not 100% sure this will be the fix but thanks to @gitKrystan @chriskrycho and @NullVoxPopuli for working through this toward a potential solution.

### CODE REVIEW

**Comment by MelSumner** [comment]
Ok it does seem like there are some undocumented pre-publish steps; working on a new release now. This reversion isn't necessary so I'm closing the PR.

**Comment by MelSumner** [comment]
so I tried running some of the scripts in the package.json and re-publish, but that didn't seem to fix it either, so looks like we might need to revert the `.npmignore` change after all.

**Comment by gitKrystan** [comment]
Looking into this. I think we need something like this in package.json:

https://github.com/emberjs/ember.js/blob/87c7fc72a3112936ae701a6768072d31caf4730d/package.json#LL173C3-L182C5

**Comment by NullVoxPopuli** [comment]
@MelSumner public-types are emitted via: https://github.com/emberjs/ember-test-helpers/blob/master/addon/types.tsconfig.json#L7

but the `prepack` command should have included that: https://github.com/emberjs/ember-test-helpers/blob/master/addon/package.json#L29 🤔

File: addon/package.json [hunk 1]
```diff
@@ -149,7 +149,7 @@
   "typesVersions": {
     "*": {
       "*": [
-        "public-types/@ember/test-helpers/*"
+        "public-types/*"
```
Comment (MelSumner) [inline addon/package.json:152]:
```suggestion
        "./public-types/*"
```

File: addon/package.json [hunk 2]
```diff
@@ -149,7 +149,7 @@
   "typesVersions": {
     "*": {
       "*": [
-        "public-types/@ember/test-helpers/*"
+        "./public-types/*"
```
Comment (MelSumner) [inline addon/package.json:152]: @gitKrystan @chriskrycho there was a tiny variation between both of your suggestion, should it be `./public-types/*` or `public-types/*`?

Comment (gitKrystan) [reply addon/package.json:152]: Either should work.

**Review by gitKrystan — APPROVED** [review]

**Comment by MelSumner** [comment]
So even with these changes, I'm still not seeing the `public_types` folder when I run `npm pack` locally. 🤔

**Comment by MelSumner** [comment]
Changing to `dist-types` from `public-types` seems to resolve the issue, in that I now get it locally when I run `npm pack` to check. But...why? makes no real logical sense. Why would it work before but not now? 🤔

**Pull request merged**